lib/Store.spec.lua

```diff
@@ -23,6 +23,40 @@ return function()
 
 			store:destruct()
 		end)
+
+		it("should instantiate with a reducer, initial state, and middlewares", function()
+			local store = Store.new(function(state, action)
+				return state
+			end, "initial state", {})
+
+			expect(store).to.be.ok()
+			expect(store:getState()).to.equal("initial state")
+
+			store:destruct()
+		end)
+
+		it("should modify the dispatch method when middlewares are passed", function()
+			local middlewareInvokeCount = 0
+
+			local store = Store.new(function(state, action)
+				return state
+			end, "initial state", {
+				function(next)
+					return function(store, action)
+						middlewareInvokeCount = middlewareInvokeCount + 1
+						next(store, action)
+					end
+				end
+			})
+
+			expect(store.dispatch).to.never.equal(Store.dispatch)
```

Comment:
I don't think we need to assert that this is the case -- if the middleware implementation *didn't actually* override the `dispatch` method, that would satisfy what we want still!

Reply:
Good point!